components/brave_extension/extension/brave_extension/helpers/shieldsUtils.ts

```diff
@@ -77,3 +77,14 @@ export const sumAdsAndTrackers = (ads: number, trackers: number) => {
 export const mergeAdsAndTrackersResources = (ads: Array<string>, trackers: Array<string>) => {
   return [ ...ads, ...trackers ]
 }
+
+export const shieldsHasFocus = (url: string) => {
+  const devtoolsURL: string =
```

Comment:
An alternative here is to check if the currently active tab is not the tab which the Shields was opened with

Reply:
The fix you proposed in the code review would make us store all active tab content in a local state, not only the URL, since Shields inherits everything from there. It would involve an addition of a _intermediary state_ to store temporary data (using React's `state`?), possibly addition of multiple tests to cover this new flow in several different areas (`activeTab` is used in multiple places), and would involve, in a quick thought, considerably more code changes. This is a minor bug, I don't think it's worth it.

I personally don't see any reason to keep Shields open without focus, unless you're debugging with devtools, which is what this change does.

I'd agree with you if we had more bugs related to `activeTab` mismatch but I don't think we have such problem. So I'm cc'ing @karenkliu again in case we explicitly want Shields to be open even when window has no focus, which is the current behavior. I don't plan to make further editions to this PR unless UX requirements doesn't match current changes.